drivers: input: cst816s: add cst816d support

aa889788 opened the pull request:
The CST816D touch chip is similar to the CST816S, with the primary difference being the chip ID. This commit extends the existing CST816S driver to support the CST816D by adding its chip ID. There are no other modifications.

fabiobaltieri reviewed (approved):
Good but check your commit format:

https://github.com/zephyrproject-rtos/zephyr/actions/runs/10208623809/job/28260166609?pr=76617

https://docs.zephyrproject.org/latest/contribute/guidelines.html

aa889788 committed:
drivers: input: cst816s: add cst816d support

The CST816D touch chip is similar to the CST816S, with the primary
difference being the chip ID. This commit extends the existing
CST816S driver to support the CST816D by adding its chip ID. There
are no other modifications.

Signed-off-by: Shang Xiangyao <shxyke@gmail.com>

aa889788 commented:
> Good but check your commit format:
> 
> https://github.com/zephyrproject-rtos/zephyr/actions/runs/10208623809/job/28260166609?pr=76617
> 
> https://docs.zephyrproject.org/latest/contribute/guidelines.html

commit message updated
